src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -1824,6 +1886,11 @@ public void updateCertificateGroups(SecuredUUID uuid, Set<UUID> groupUuids) thro
     public void updateOwner(SecuredUUID uuid, String ownerUuid) throws NotFoundException {
         Certificate certificate = getCertificateEntityWithAssociations(uuid);
 
+        if (certificate.isArchived()) {
+            logger.info("Certificate with UUID {} is archived and owner will not be updated.", uuid);
```

Comment:
Here is more suitable to throw Exception since it is user triggered operation that is not allowed. It should say it is not allowed to change owner for archived certificate.